python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt_tests/test_assets.py

```diff
@@ -6,87 +6,154 @@
 from dagster import AssetSpec, Definitions, file_relative_path
 from dagster._core.definitions import build_assets_job
 from dagster_embedded_elt.sling import SlingMode, SlingResource, build_sling_asset
-from dagster_embedded_elt.sling.resources import SlingSourceConnection, SlingTargetConnection
+from dagster_embedded_elt.sling.resources import (
+    SlingSourceConnection,
+    SlingTargetConnection,
+)
+
+
+@pytest.fixture
+def test_csv():
+    return os.path.abspath(file_relative_path(__file__, "test.csv"))
+
+
+@pytest.fixture
+def temp_db():
+    with tempfile.TemporaryDirectory() as tmpdir_path:
+        dbpath = os.path.join(tmpdir_path, "sqlite.db")
+        yield dbpath
+
+
+@pytest.fixture
+def sling_sqlite_resource(temp_db):
+    return SlingResource(
+        source_connection=SlingSourceConnection(type="file"),
+        target_connection=SlingTargetConnection(
+            type="sqlite", connection_string=f"sqlite://{temp_db}"
+        ),
+    )
+
+
+ASSET_SPEC = AssetSpec(
+    key=["main", "tbl"],
+    group_name="etl",
+    description="ETL Test",
+    deps=["foo"],
+)
 
 
 @pytest.mark.parametrize(
     "mode,runs,expected", [(SlingMode.INCREMENTAL, 1, 3), (SlingMode.SNAPSHOT, 2, 6)]
 )
-def test_build_sling_asset(mode: SlingMode, runs: int, expected: int):
-    with tempfile.TemporaryDirectory() as tmpdir_path:
-        fpath = os.path.abspath(file_relative_path(__file__, "test.csv"))
-        dbpath = os.path.join(tmpdir_path, "sqlite.db")
+def test_build_sling_asset(
+    test_csv: str,
+    temp_db: str,
+    sling_sqlite_resource: SlingResource,
+    mode: SlingMode,
+    runs: int,
+    expected: int,
+):
+    asset_def = build_sling_asset(
+        asset_spec=ASSET_SPEC,
+        source_stream=f"file://{test_csv}",
+        target_object="main.tbl",
+        mode=mode,
+        primary_key="SPECIES_CODE",
+        sling_resource_key="sling_resource",
```

Comment:
Any reason why we're using `sling_resource` here instead of relying on the default `sling`?

Reply:
should be a separate test then, just to keep the code for other tests simple.